inc/class-module.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * The Module object provides a common interface for registering
+ * modules and carrying out module related tasks.
+ *
+ * @package hm-platform
+ */
+
+namespace HM\Platform;
+
+class Module {
+
+	/**
+	 * Store of all registered modules.
+	 *
+	 * @var array
+	 */
+	protected static $modules = [];
+
+	/**
+	 * The slug for the module, used to reference individual modules.
+	 *
+	 * @var string
+	 */
+	protected $slug;
```

Comment:
I think we need an accessor for `slug` as well?

Reply:
Sure, I'd left it out because that's the array key/accessor, so presumably you'd have the slug already.